tests/main/refresh-devmode-nonproduction/task.yaml

```diff
@@ -0,0 +1,30 @@
+summary: Check that the refresh command works.
+
+systems: [-ubuntu-core-16-64, -ubuntu-core-16-device-amd64, -ubuntu-core-16-device-arm]
+
+details: |
+    These tests exercise the refresh command using different store backends.
+    The concrete store to be used is controlled with the STORE_TYPE environment
+    vairiable, if it is empty, a fake local store is used, if it has a value of
+    staginig or production the corresponding remote store is used.
+    When executing against the remote stores the tests rely in the existence of
+    a given snap with an updatable version (version string like 2.0+fake1) in the
+    edge channel.
+
+environment:
+    SNAP_NAME: test-snapd-tools
+    SNAP_VERSION_PATTERN: \d+\.\d+\+fake1
+    BLOB_DIR: $(pwd)/fake-store-blobdir
+    STORE_TYPE: fake
```

Comment:
This is moving the logic of the task out into `tasks/refresh.sh`, apparently to avoid duplicating it across tasks, but the tasks are all being duplicated to use it. Feels like it's precisely creating the problem it is intended to solve.

I bet we can have one single test file, named simply tests/main/refresh/task.yaml, which contains the real logic inside it, and which has a few variants being executed inside it.

Please ping me if you find any issues in doing so.